lib/iris/util.py

```diff
@@ -2189,3 +2190,127 @@ def mask_cube_from_shapefile(cube, shape, minimum_weight=0.0, in_place=False):
     masked_cube = mask_cube(cube, shapefile_mask, in_place=in_place)
     if not in_place:
         return masked_cube
+
+
+def equalise_cubes(
+    cubes,
+    apply_all=False,
+    unify_names=False,
+    equalise_attributes=False,
+    unify_time_units=False,
+):
+    """Modify a set of cubes to assist merge/concatenate operations.
+
+    Various different adjustments can be applied to the input cubes, to remove
+    differences which may prevent them from combining into larger cubes.  The requested
+    "equalisation" operations are applied to each group of input cubes with matching
+    cube metadata (names, units, attributes and cell-methods).
+
+    Parameters
+    ----------
+    cubes : sequence of :class:`~iris.cube.Cube`
+        The input cubes, in a list or similar.
+
+    apply_all : bool, default=False
+        Enable *all* the equalisation operations.
+
+    unify_names : bool, default=False
+        When True, remove any redundant ``var_name`` and ``long_name`` properties,
+        leaving only one ``standard_name``, ``long_name`` or ``var_name`` per cube.
+        In this case, the adjusted names are also used when selecting input groups.
+
+    equalise_attributes : bool, default=False
+        When ``True``, apply an :func:`equalise_attributes` operation to each input
+        group.  In this case, attributes are ignored when selecting input groups.
+
+    unify_time_units : bool, default=False
+        When True, apply the :func:`unify_time_units` operation to each input group.
+        Note : while this may convert units of time reference coordinates, it does
+        not affect the units of the cubes themselves.
+
+    Returns
+    -------
+    :class:`~iris.cube.CubeList`
+        A CubeList containing the original input cubes, ready for merge or concatenate
+        operations.  The cubes are possibly modified (in-place), and possibly in a
+        different order.
+
+    Notes
+    -----
+    All the 'equalise' operations operate in a similar fashion, in that they identify
+    and remove differences in a specific metadata element, altering metadata so that
+    a merge or concatenate can potentially combine a group of cubes into a single
+    result cube.
+
+    The various 'equalise' operations are not applied to the entire input, but to
+    groups of input cubes with the same ``cube.metadata``.
+
+    The input cube groups also depend on the equalisation operation(s) selected :
+    Operations which equalise a specific cube metadata element (names, units,
+    attributes or cell-methods) exclude that element from the input grouping criteria.
+
+    """
+    from iris.common.metadata import CubeMetadata
+    from iris.cube import CubeList
+
+    if unify_names or apply_all:
```

Comment:
Is `unify_names` the correct name for this operation? Compared to `unify_time_units`, this operation works quite differently. Only removing information rather than editing it. `unify_names` might suggest that equivalent `standard_name`s are made the same for example. This may also cause us problems if we ever wanted to add that sort of functionality in the future. A better name might be `standardise_names` or `trim_names`.

Reply:
now renamed to 'normalise_names' : https://github.com/SciTools/iris/pull/6257/commits/844b18d99bda75e54031c341002c94944ba0c256

I didn't like "standardise_names" because of the crossover with "standard_name"